plugin/src/main/groovy/com/btkelly/gnag/utils/ReportWriter.java

```diff
@@ -42,29 +41,27 @@
     private static final String HTML_REPORT_SUFFIX =
             "</article></html>";
 
-    public static boolean writeReportToDirectory(
+    public static boolean writeLocalReportFiles(
             @NotNull final Set<Violation> violations,
             @NotNull final File directory) {
+        
+        if (violations.isEmpty()) {
+            throw new IllegalStateException("This method should only be called when violations were detected.");
```

Comment:
It's intended to be a developer warning only - it's considered an error to call this function without first checking for non-empty violations. Does that change your mind?